engine/src/main/java/org/pentaho/di/trans/steps/setvalueconstant/SetValueConstant.java

```diff
@@ -79,37 +79,36 @@ public boolean processRow( StepMetaInterface smi, StepDataInterface sdi ) throws
       data.setConvertRowMeta( data.getOutputRowMeta().cloneToType( ValueMetaInterface.TYPE_STRING ) );
 
       // Consider only selected fields
```

Comment:
@bmorrise 
You call meta.getFields() and meta.getFields().size() quite a few times. Piggybacking off of the "anti-patterns" meeting, maybe you can call them each once when you enter "if ( !Utils.isEmpty( meta.getFields() ) ) {" 

i.e.
if ( !Utils.isEmpty( meta.getFields() ) ) {
Field fields = meta.getFields();
int size = fields.size();
.....
}